scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala

```diff
@@ -0,0 +1,160 @@
+    * @tparam T serializable type
+    */
+  case class JdbcReadOptions[T](dbConnectionOptions: DbConnectionOptions,
+                                query: String,
+                                statementPreparator: (PreparedStatement) => Unit,
+                                rowMapper: (ResultSet) => T)
+
+  /**
+    * Values need to initiate write connection to database.
+    *
+    * @param dbConnectionOptions Options need to create a database connection.
+    * @param statement JDBC query statement
+    * @param preparedStatementSetter
+    * @tparam T serializable type
+    */
+  case class JdbcWriteOptions[T](dbConnectionOptions: DbConnectionOptions,
+                                 statement: String,
+                                 preparedStatementSetter: (T, PreparedStatement) => Unit)
+
+
+  case class JdbcSqlIO[T](table: String) extends TestIO[T](table)
+
+  /** Enhanced version of [[ScioContext]] with Jdbc and Cloud SQL methods. */
+  implicit class JdbcScioContext(@transient val self: ScioContext) extends Serializable {
+    /**
+      * Get an SCollection for a CloudSql query
+      *
+      * @group input
+      */
+    def cloudSqlSelect[T: ClassTag](readOptions: JdbcReadOptions[T])
+    : SCollection[T] = self.requireNotClosed {
+      if (self.isTest) {
+        self.getTestInput(JdbcSqlIO[T](TEST_READ_TABLE_NAME))
+      } else {
+        jdbcSelect[T](readOptions)
+      }
+    }
+
+    /**
+      * Get as SCollection for JDBC query
+      *
+      * @group input
+      */
+    def jdbcSelect[T: ClassTag](readOptions: JdbcReadOptions[T])
+    : SCollection[T] = self.requireNotClosed {
+      if (self.isTest) {
+        self.getTestInput(JdbcSqlIO[T](TEST_READ_TABLE_NAME))
+      } else {
+        val coder = self.pipeline.getCoderRegistry.getScalaCoder[T]
+        val conOpt = readOptions.dbConnectionOptions
+        val transformer = JdbcIO.read[T]()
+          .withCoder(coder)
+          .withDataSourceConfiguration(DataSourceConfiguration
+            .create(conOpt.driverClass.getCanonicalName, conOpt.connectionUrl)
+            .withUsername(conOpt.username)
+            .withPassword(conOpt.password))
+          .withQuery(readOptions.query)
+          .withStatementPrepator(new StatementPreparator {
+            override def setParameters(preparedStatement: PreparedStatement): Unit = {
+              readOptions.statementPreparator(preparedStatement)
+            }
+          })
+          .withRowMapper(new RowMapper[T] {
+            override def mapRow(resultSet: ResultSet): T = {
+              readOptions.rowMapper(resultSet)
+            }
+          })
+        self.wrap(self.applyInternal(transformer)).setName(self.tfName)
+      }
+    }
+  }
+
+  /**
+    * Enhanced version of [[com.spotify.scio.values.SCollection SCollection]] with JDBC and
+    * Cloud SQL methods.
+    */
+  implicit class JdbcSCollection[T](val self: SCollection[T]) extends Serializable {
```

Comment:
Is `JdbcSCollection` need to be `Serializable`, do you need to mark `self` as `@transient`?